src/components/File.jsx

```diff
@@ -34,6 +35,26 @@ class File extends Component {
     }
   }
 
+  componentDidMount () {
+    this.gesturesHandler = new Hammer.Manager(this.fil)
+    this.gesturesHandler.add(new Hammer.Tap({ event: 'doubletap', taps: 2 }))
+    this.gesturesHandler.add(new Hammer.Tap({ event: 'singletap' }))
+    this.gesturesHandler.add(new Hammer.Press({ event: 'onpress' }))
+    this.gesturesHandler.get('doubletap').recognizeWith('singletap').requireFailure('onpress')
+    this.gesturesHandler.get('singletap').requireFailure('doubletap').requireFailure('onpress')
+    this.gesturesHandler.on('onpress singletap doubletap', (ev) => {
+      if (ev.type === 'onpress' || (this.props.selectionModeActive && ev.type === 'singletap')) {
+        this.toggle(ev.srcEvent)
+      } else {
+        this.open(ev.srcEvent, this.props.attributes)
+      }
+    })
```

Comment:
The code may be more explicit by using a named function like `enableTouchEvents`.